kitsune/sumo/handlers.py

```diff
@@ -0,0 +1,42 @@
+from abc import ABC, abstractmethod
+from collections.abc import Sequence
+from dataclasses import dataclass, field
+from typing import Generic, TypeVar
+
+T = TypeVar("T")
+
+
+class AbstractHandler(ABC):
+    """Abstract base class for all handlers."""
+
+    @abstractmethod
+    def handle(self) -> None:
+        pass
+
+
+@dataclass
+class AbstractChain(Generic[T]):
```

Comment:
I don't fully understand all of `typing` stuff, so I'm probably missing something, but I would have expected `AbstractHandler` instead `T` within this class.